### PR TITLE
Sync v2-6-stable with v2-6-test to release 2.6.1

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Time for `2.6.1rc1`!

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
LGTM. I have one  doubt about https://github.com/apache/airflow/pull/30352 - it has new configuraiton param on one hand, but it does solve a problem where  redis sentinel configuration was broken. So I would be in getting it in as it looks pretty safe/

**Review by potiuk — APPROVED** [review]
LGTM. I have one  doubt about https://github.com/apache/airflow/pull/30352 - it has new configuraiton param on one hand, but it does solve a problem where  redis sentinel configuration was broken. So I would be in getting it in as it looks pretty safe/

**Comment by eladkal** [comment]
https://github.com/apache/airflow/pull/31125 is missing from the release

**Comment by ephraimbuddy** [comment]
> #30352

Let me move it to 2.7.0. since it added a config. I accessed it by the issue it closed and didn't look the code change. It should be 2.7.0

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,7 +21,53 @@
...
+- Fix problem with displaying graph (#30765)
+- Fix backfill KeyError when try_number out of sync (#30653)
+- Re-enable clear on TaskInstanceModelView for role User (#30415)
+- Fix DAG.access_control can't sync when clean access_control (#30340)
+- Prevent DagRun's ``start_date`` from reset (#30124) (#30125)
+
+Misc/Internal
+"""""""""""""
+- Present affected task instances as table (#30633)
+- Fix missing models in API documentation. (#31021)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Show task instances affected by clearing in a table (#30633)
- Fix missing models in API documentation (#31021)
```

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -21,7 +21,53 @@
...
+
+Doc only changes
+""""""""""""""""
+- Improve Description of dag_processing.processes Metric in Docs (#30891)
+- Improve Quick Start instructions (#30820)
+- Update faq.rst about missing task logs (#30717)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Add section about missing task logs to the FAQ (#30717)
```

File: RELEASE_NOTES.rst [hunk 3]
```diff
@@ -21,7 +21,53 @@
...
+- Prevent DagRun's ``start_date`` from reset (#30124) (#30125)
+
+Misc/Internal
+"""""""""""""
+- Present affected task instances as table (#30633)
+- Fix missing models in API documentation. (#31021)
+
+Doc only changes
+""""""""""""""""
+- Improve Description of dag_processing.processes Metric in Docs (#30891)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Improve description of the ``dag_processing.processes`` metric (#30891)
```

File: RELEASE_NOTES.rst [hunk 4]
```diff
@@ -21,7 +21,53 @@
...
+- Improve Quick Start instructions (#30820)
+- Update faq.rst about missing task logs (#30717)
+- docker compose doc changes (#30662)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Mount the `config` directory in docker compose (#30662)
```

File: RELEASE_NOTES.rst [hunk 5]
```diff
@@ -21,7 +21,53 @@
...
+- Fix ``order_by`` request in list DAG rest api (#30926)
+- Include node height/width in center-on-task logic (#30924)
+- Remove print from dag trigger command (#30921)
+- Improve task group UI in new graph (#30918)
+- Fix mapped states case (#30916)
+- Fix problem with displaying graph (#30765)
+- Fix backfill KeyError when try_number out of sync (#30653)
+- Re-enable clear on TaskInstanceModelView for role User (#30415)
+- Fix DAG.access_control can't sync when clean access_control (#30340)
+- Prevent DagRun's ``start_date`` from reset (#30124) (#30125)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Prevent DagRun's ``state`` and ``start_date`` from being reset when clearing a task in a running DagRun (#30125)
```

File: README.md [hunk 6]
```diff
@@ -87,15 +87,15 @@ Airflow is not a streaming solution, but it is often used to process real-time d
...
-| Kubernetes          | 1.23, 1.24, 1.25, 1.26       | 1.21, 1.22, 1.23, 1.24, 1.25 |
-| PostgreSQL          | 11, 12, 13, 14, 15           | 11, 12, 13, 14, 15           |
-| MySQL               | 5.7, 8                       | 5.7, 8                       |
-| SQLite              | 3.15.0+                      | 3.15.0+                      |
-| MSSQL               | 2017(\*), 2019(\*)           | 2017(\*), 2019(\*)           |
+|                     | Main version (dev)           | Stable version (2.6.1)             |
+|---------------------|------------------------------|------------------------------------|
+| Python              | 3.7, 3.8, 3.9, 3.10          | 3.7, 3.8, 3.9, 3.10                |
+| Platform            | AMD64/ARM64(\*)              | AMD64/ARM64(\*)                    |
+| Kubernetes          | 1.23, 1.24, 1.25, 1.26       | 1.21, 1.22, 1.23, 1.24, 1.25, 1.26 |
```
Comment (raphaelauv) [inline README.md:94]: you should cherry-pick this commit https://github.com/apache/airflow/pull/31085

File: RELEASE_NOTES.rst [hunk 7]
```diff
@@ -21,7 +21,53 @@
...
+- Add support for dynamic tasks with template fields that contain ``pandas.DataFrame`` (#30943)
+- Use the Task Group explicitly passed to 'partial' if any (#30933)
+- Fix ``order_by`` request in list DAG rest api (#30926)
+- Include node height/width in center-on-task logic (#30924)
+- Remove print from dag trigger command (#30921)
+- Improve task group UI in new graph (#30918)
+- Fix mapped states case (#30916)
+- Fix problem with displaying graph (#30765)
+- Fix backfill KeyError when try_number out of sync (#30653)
+- Re-enable clear on TaskInstanceModelView for role User (#30415)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Re-enable clear and setting state in the TaskInstance UI (#30415)
```

File: RELEASE_NOTES.rst [hunk 8]
```diff
@@ -21,7 +21,53 @@
...
+- Fix DAG.access_control can't sync when clean access_control (#30340)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: I have a concern with this change. I think it fixes one issue (empty access_control isn't respected), but creates another (it wipes out any rights not set by access_control initially).

File: RELEASE_NOTES.rst [hunk 9]
```diff
@@ -21,7 +21,53 @@
...
+- Fix ``airflow providers get`` command output (#30978)
+- Fix Pool schema OpenAPI spec (#30973)
+- Update incorrect config values (#30969)
+- Add support for dynamic tasks with template fields that contain ``pandas.DataFrame`` (#30943)
+- Use the Task Group explicitly passed to 'partial' if any (#30933)
+- Fix ``order_by`` request in list DAG rest api (#30926)
+- Include node height/width in center-on-task logic (#30924)
+- Remove print from dag trigger command (#30921)
+- Improve task group UI in new graph (#30918)
+- Fix mapped states case (#30916)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix mapped states in grid view (#30916)
```

File: RELEASE_NOTES.rst [hunk 10]
```diff
@@ -21,7 +21,53 @@
...
+- fix MappedTaskGroup import in taskinstance (#31100)
+- Format DagBag.dagbag_report() Output (#31095)
+- Fix template error when iterating None value and fix params documentation (#31078)
+- fix EXTRAS_DEPENDENCIES of apache-hive (#31068)
+- Fix issue in pre-importing modules in zipfile (#31061)
+- Move TaskInstanceKey to a separate file (#31033)
+- Fix unable to remove DagRun and TaskInstance with note (#30987)
+- Fix ``airflow providers get`` command output (#30978)
+- Fix Pool schema OpenAPI spec (#30973)
+- Update incorrect config values (#30969)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: This feels like a docs-only change really. We should update the wording too.

File: RELEASE_NOTES.rst [hunk 11]
```diff
@@ -21,7 +21,53 @@
...
+- Move interleave_timestamp_parser config to section logging (#31102)
+- fix MappedTaskGroup import in taskinstance (#31100)
+- Format DagBag.dagbag_report() Output (#31095)
+- Fix template error when iterating None value and fix params documentation (#31078)
+- fix EXTRAS_DEPENDENCIES of apache-hive (#31068)
+- Fix issue in pre-importing modules in zipfile (#31061)
+- Move TaskInstanceKey to a separate file (#31033)
+- Fix unable to remove DagRun and TaskInstance with note (#30987)
+- Fix ``airflow providers get`` command output (#30978)
+- Fix Pool schema OpenAPI spec (#30973)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix Pool schema in the OpenAPI spec (#30973)
```

File: RELEASE_NOTES.rst [hunk 12]
```diff
@@ -21,7 +21,53 @@
...
+Bug Fixes
+^^^^^^^^^
+- Move interleave_timestamp_parser config to section logging (#31102)
+- fix MappedTaskGroup import in taskinstance (#31100)
+- Format DagBag.dagbag_report() Output (#31095)
+- Fix template error when iterating None value and fix params documentation (#31078)
+- fix EXTRAS_DEPENDENCIES of apache-hive (#31068)
+- Fix issue in pre-importing modules in zipfile (#31061)
+- Move TaskInstanceKey to a separate file (#31033)
+- Fix unable to remove DagRun and TaskInstance with note (#30987)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix deleting DagRuns and TaskInstances that have a note (#30987)
```

File: RELEASE_NOTES.rst [hunk 13]
```diff
@@ -21,7 +21,53 @@
...
+
+Bug Fixes
+^^^^^^^^^
+- Move interleave_timestamp_parser config to section logging (#31102)
+- fix MappedTaskGroup import in taskinstance (#31100)
+- Format DagBag.dagbag_report() Output (#31095)
+- Fix template error when iterating None value and fix params documentation (#31078)
+- fix EXTRAS_DEPENDENCIES of apache-hive (#31068)
+- Fix issue in pre-importing modules in zipfile (#31061)
+- Move TaskInstanceKey to a separate file (#31033)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Move TaskInstanceKey to a separate file to fix circular import (#31033)
```

File: RELEASE_NOTES.rst [hunk 14]
```diff
@@ -21,7 +21,53 @@
...
+No significant changes.
+
+Bug Fixes
+^^^^^^^^^
+- Move interleave_timestamp_parser config to section logging (#31102)
+- fix MappedTaskGroup import in taskinstance (#31100)
+- Format DagBag.dagbag_report() Output (#31095)
+- Fix template error when iterating None value and fix params documentation (#31078)
+- fix EXTRAS_DEPENDENCIES of apache-hive (#31068)
+- Fix issue in pre-importing modules in zipfile (#31061)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix issue with zipfiles in DAGs folder when pre-importing Airflow modules (#31061)
```

File: RELEASE_NOTES.rst [hunk 15]
```diff
@@ -21,7 +21,53 @@
...
+^^^^^^^^^^^^^^^^^^^
+No significant changes.
+
+Bug Fixes
+^^^^^^^^^
+- Move interleave_timestamp_parser config to section logging (#31102)
+- fix MappedTaskGroup import in taskinstance (#31100)
+- Format DagBag.dagbag_report() Output (#31095)
+- Fix template error when iterating None value and fix params documentation (#31078)
+- fix EXTRAS_DEPENDENCIES of apache-hive (#31068)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix ``apache-hive`` extra so it installs the correct package (#31068)
```

File: RELEASE_NOTES.rst [hunk 16]
```diff
@@ -21,7 +21,53 @@
...
+--------------------------
+
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+No significant changes.
+
+Bug Fixes
+^^^^^^^^^
+- Move interleave_timestamp_parser config to section logging (#31102)
+- fix MappedTaskGroup import in taskinstance (#31100)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix ``MappedTaskGroup`` import in taskinstance file (#31100)
```

File: RELEASE_NOTES.rst [hunk 17]
```diff
@@ -21,7 +21,53 @@
...
+Airflow 2.6.1 (2023-05-12)
+--------------------------
+
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+No significant changes.
+
+Bug Fixes
+^^^^^^^^^
+- Move interleave_timestamp_parser config to section logging (#31102)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Move ``interleave_timestamp_parser`` config to the logging section (#31102)
```

**Review by potiuk — CHANGES_REQUESTED** [review]
Also - should not we also include that one? https://github.com/apache/airflow/pull/31101 ? I think that one is pretty important to get merged @dstandish @ephraimbuddy ? - a number of people have hard time finding no logs in their tasks.

File: RELEASE_NOTES.rst (hunk 8 above)
Comment (ephraimbuddy) [reply RELEASE_NOTES.rst]: Then I will drop it

**Comment by ashb** [comment]
Please hold off - I need to look at a possible issue with log redaction first. I will do that tomorrow (was public holiday in UK today)

**Comment by potiuk** [comment]
> Please hold off - I need to look at a possible issue with log redaction first. I will do that tomorrow (was public holiday in UK today)

:crown:

**Comment by ashb** [comment]
(Hmmm, no guillotine emoji. Shame)

**Comment by dstandish** [comment]
I just merged #31101 and set its milestone back to 2.6.1 fyi

**Comment by ashb** [comment]
Here's my PR https://github.com/apache/airflow/pull/31149

**Review by potiuk — APPROVED** [review]

**Comment by herlambang** [comment]
> #31125 is missing from the release

this is not in release notes yet, or it shouldn't?

File: RELEASE_NOTES.rst [hunk 18]
```diff
@@ -21,7 +21,55 @@
 
 .. towncrier release notes start
 
-Airflow 2.6.0 (2023-04-28)
+Airflow 2.6.1 (2023-05-12)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
Airflow 2.6.1 (2023-05-15)
```

Let's do the 15th so we aren't releasing over the weekend again.

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**